internal/net/http/client/option_test.go

```diff
@@ -22,1945 +22,1378 @@ import (
-			       test.checkFunc = defaultCheckFunc
-			   }
-			   got := WithEnableCompression(test.args.enable)
-			   obj := new(T)
-			   got(obj)
-			   if err := test.checkFunc(test.want, obj); err != nil {
-			       tt.Errorf("error = %v", err)
-			   }
-			*/
+			got := WithEnableCompression(test.args.enable)
+			obj := &T{
+				Transport: &http.Transport{},
+			}
+			if err := test.checkFunc(test.want, obj, got(obj)); err != nil {
+				tt.Errorf("error = %v", err)
+			}
 		})
 	}
 }
 
 func TestWithMaxIdleConns(t *testing.T) {
-	// Change interface type to the type of object you are testing
-	type T = interface{}
+	type T = transport
 	type args struct {
 		cn int
 	}
 	type want struct {
 		obj *T
-		// Uncomment this line if the option returns an error, otherwise delete it
-		// err error
+		err error
 	}
 	type test struct {
-		name string
-		args args
-		want want
-		// Use the first line if the option returns an error. otherwise use the second line
-		// checkFunc  func(want, *T, error) error
-		// checkFunc  func(want, *T) error
+		name       string
+		args       args
+		want       want
+		checkFunc  func(want, *T, error) error
 		beforeFunc func(args)
 		afterFunc  func(args)
 	}
-
-	// Uncomment this block if the option returns an error, otherwise delete it
-	/*
-	   defaultCheckFunc := func(w want, obj *T, err error) error {
-	       if !errors.Is(err, w.err) {
-	           return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
-	       }
-	       if !reflect.DeepEqual(obj, w.obj) {
-	           return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", obj, w.obj)
-	       }
-	       return nil
-	   }
-	*/
-
-	// Uncomment this block if the option do not returns an error, otherwise delete it
-	/*
-	   defaultCheckFunc := func(w want, obj *T) error {
-	       if !reflect.DeepEqual(obj, w.obj) {
-	           return errors.Errorf("got: \"%#v\",\n\t\t\t\twant: \"%#v\"", obj, w.obj)
-	       }
-	       return nil
-	   }
-	*/
+	defaultCheckFunc := func(w want, obj *T, err error) error {
+		if !errors.Is(err, w.err) {
+			return errors.Errorf("got_error: \"%#v\",\n\t\t\t\twant: \"%#v\"", err, w.err)
+		}
+		if diff := comparator.Diff(obj, w.obj, transportComparator...); diff != "" {
+			return errors.New(diff)
+		}
+		return nil
+	}
 
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           cn: 0,
-		       },
-		       want: want {
-		           obj: new(T),
-		       },
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           cn: 0,
-		           },
-		           want: want {
-		               obj: new(T),
-		           },
-		       }
-		   }(),
-		*/
+		{
+			name: "set conn success",
+			args: args{
+				cn: 5,
+			},
+			want: want{
+				obj: &T{
+					Transport: &http.Transport{
+						MaxIdleConns: 5,
+					},
+				},
+			},
+		},
```

Comment:
I think it'd be better when the case of no value will be created.
```go
{
  name: "set conn success with no input"
  want: want{
    obj: &T{
      Transport: &http.Transport{
        MacIdelConns: 0,
      }
    }
  }
}

Reply:
@vankichi In golang, the default value of `int` is 0, so if we write 0 here, we can not differentiate if the value is set or not, it I think this test case may be meaningless.
What do you think?

Reply:
I think it will be set as 0 when the no value is input is not meaningless and better for test coverage.
I think `set 0 with no input`, which means no logic will be run,  is the one of the test  case.

Reply:
I will fix it, thank you for your opinion